rir/src/compiler/opt/scope_resolution.cpp

```diff
@@ -128,16 +129,31 @@ class TheScopeResolution {
 
             std::unordered_map<BB*, Value*> inputs;
             bool fail = false;
+            Value* initialValue = nullptr;
```

Comment:
initialValue? Isn't this only here for unbound?

Reply:
well, again, it would work for any initial value... e.g.:

```
mkenv(x=true)
if (...)
  x <- 1
```

here we have `{val=true, origin=nullptr} | {val = 1, origin=x<-1}` as the abstract state

Reply:
actually no, in this case we set the mkenv as the origin. you convinced me. this should only happen with unbound.